src/helm/benchmark/slurm_runner.py

```diff
@@ -0,0 +1,220 @@
+import dataclasses
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Dict, List
+import argparse
+import json
+import os
+import time
+import sys
+
+import dacite
+
+from helm.benchmark.executor import ExecutionSpec
+from helm.benchmark.runner import Runner, RunSpec, RunnerError
+from helm.benchmark.slurm_jobs import (
+    submit_slurm_job,
+    cancel_slurm_job,
+    get_slurm_job_state,
+    SlurmJobState,
+    TERMINAL_SLURM_JOB_STATES,
+    FAILURE_SLURM_JOB_STATES,
+)
+from helm.common.general import ensure_directory_exists, asdict_without_nones
+from helm.common.hierarchical_logger import hlog
+
+
+@dataclass
+class SlurmRunnerSpec:
+    """Arguments to instantiate a SlurmRunner."""
+
+    execution_spec: ExecutionSpec
+    output_path: str
+    suite: str
+    skip_instances: bool
+    cache_instances: bool
+    cache_instances_only: bool
+    skip_completed_runs: bool
+    exit_on_error: bool
+
+    def to_kwargs(self):
+        return {field.name: getattr(self, field.name) for field in dataclasses.fields(self)}
+
+
+class SlurmRunner(Runner):
+    """Runner that runs the entire benchmark on Slurm."""
```

Comment:
Would be nice to have a high-level description of what the flow is.

Reply:
Added flow description to `run_all()`.